Tests/SonarScanner.MSBuild.Shim.Test/PropertiesFileGeneratorTests.cs

```diff
@@ -418,15 +455,55 @@ public void FileGen_FilesOutsideProjectPath()
             var result = new PropertiesFileGenerator(config, logger).GenerateFile();
 
             // Assert
-            AssertExpectedStatus("projectName", ProjectInfoValidity.NoFilesToAnalyze, result);
+            AssertExpectedStatus("project", ProjectInfoValidity.NoFilesToAnalyze, result);
             AssertExpectedProjectCount(1, result);
-
             // No files -> project file not created
             AssertFailedToCreatePropertiesFiles(result, logger);
+            logger.AssertWarningsLogged(1);
+            logger.AssertSingleWarningExists($"File '{fileOutsideRootPath}' is not located under the root directory");
         }
 
         [TestMethod]
-        public void FileGen_SharedFiles()
+        public void GenerateFile_FilesOutOfProjectPath_PrintsCorrectWarnings()
```

Comment:
`GenerateFile_FileOutsideProjectPath_IsNotAnalyzedAndWarningIsLogged` tests that the file which is ignored for being out of the root directory is not analyzed at all.
I also added the extra assert to check that the correct warning is logged (because before the test was passing but not because the file was ignored due to being out of the root dir, but because it did not exist at all).

`GenerateFile_FileOutsideProjectPath_IsNotAnalyzedAndWarningIsLogged` tests that there are no warnings for the files that we want to exclude from creating noise (exe, DLL and in NuGet cache).


Indeed it makes sense to merge them.

Reply:
I did not suggest to merge, but to rename the second one